sdk-lib/src/main/java/cash/z/ecc/android/sdk/internal/model/Proposal.kt

```diff
@@ -0,0 +1,21 @@
+package cash.z.ecc.android.sdk.internal.model
+
+import cash.z.ecc.android.sdk.model.Zatoshi
+
+class Proposal(
+    private val inner: ProposalUnsafe
+) {
+    companion object {
+        fun parse(proposal: ProposalUnsafe): Proposal {
+            return Proposal(proposal)
```

Comment:
Is the intent here to just create a wrapper so that the underlying `ProposalUnsafe` is not mutable, and that `Proposal` is all that is exposed in public APIs?

I don't understand why this method is called `parse`, given that it doesn't appear to do any parsing or validation. Also, given that `toUnsafe` is exposed, I'm not sure what this class is supposed to be doing.

Reply:
The `ProposalUnsafe` class only exists because of the split between `backend-lib` and `sdk-lib`. I cannot access the type-safe types like `Zatoshi` from inside `backend-lib`, but I also don't want to expose the proposal protobuf outside `backend-lib`. So this `Proposal.parse` function is really the only one I want, and we will need to do some safety checks here depending on what we want to expose, but for now I'll rename it to something else.